### DISCUSSION TITLE
How important is Data/Read?

### DISCUSSION THREAD

**Discussion by boxgaming** [boxgaming/qbjs#10] [Polls]
Let us know your opinion:

**Comment by ghost** [comment]
To be honest, I've used it only once and with a very specific kind of program. However, I don't deal with graphics and I feel like it is mostly used by those who do, if I'm not mistaken.

**Comment by felixp7** [comment]
While my first Basic didn't have it, DATA is vital for making data-driven games like text adventures or roguelikes. I've been able to get by with array literals instead, in dialects that have them, but it's not the same thing.

**Comment by mobluse** [comment]
I think DATA, READ, and RESTORE are important for compatibility with earlier BASIC programs. I believe QBasic is backwards compatible with GW-BASIC.

**Comment by Pinebelly** [comment]
I am an hobbyist technologist and have looked over retro code implementing data/read structures used in some of the old 8-bit machines. @mobluse seems to have a valid point that porting older code to qbjs is made easier but also would having data/read structures have educational value by giving a comparative basis for understanding the evolution of data structures through the decades? Therefore, can implementing data/read/restore help to continue the educational utility of preserving archaic data structures for those who study programming? Qbjs has incredible potential to be a preservation tool of early BASIC programming with its ease of use in browsers.

**Reply by boxgaming** [reply]
Thanks everyone who responded to this post.  I should have posted an update to this poll some time ago. Support for Data, Read and Restore was added in release 0.5.0.
